eder/rails-sportsbook
# ## Implementing and Documenting the Sports Management API

### Description:

This Pull Request introduces the sports management functionality as specified in the project requirements. Here's a breakdown of the primary changes:

- **Database Modeling:** 
  - Introduced the `Sport` model with fields `name`, `slug`, and `active`.
  - Created migrations to structure the database accordingly.

- **API Endpoints:**
  - **POST** `/api/v1/sports`: For creating a new sport.
  - **GET** `/api/v1/sports`: For retrieving all sports.
  - **GET** `/api/v1/sports/{id}`: To fetch details of a specific sport.
  - **PUT** `/api/v1/sports/{id}`: To update a specific sport.
  - **DELETE** `/api/v1/sports/{id}`: To delete a specific sport.

- **Tests:**
  - Added RSpec tests for the `Sport` model ensuring validations and behaviors.
  - Introduced integration tests for all the above endpoints.

- **Documentation:**
  - Integrated Swagger for API documentation.
  - Crafted Swagger documentation for all sports-related endpoints.

- **Refactoring & Enhancements:**
  - Modified the controller structure to fit into the `Api::V1` module pattern.
  - Adjustments made in routes to mirror the modular structure.
  - Employed `parameterize` for the automatic generation of the `slug` field based on `name`.

### Review Instructions:

1. Review the migrations and model to ensure correct structure and validations.
2. Check the implementation of endpoints against specified requirements.
3. Run the RSpec tests to ensure everything behaves as expected.
4. Navigate through the Swagger documentation ensuring all endpoints are well-documented.

### Testing Instructions:

1. Execute `bundle install` to install all necessary gems.
2. Run `rails db:migrate` to apply the migrations.
3. Start the Rails server with `rails s`.
4. Access the Swagger documentation at [http://localhost:3000/api-docs](http://localhost:3000/api-docs) to explore and test the sports API endpoints.
